### ISSUE TITLE
Attempting to enable federation of an already-enabled type should fail fast

### ISSUE THREAD

**Issue by marun** [kubernetes-sigs/federation-v2#743]
Attempting to enable federation of an already-enabled type with a non-default group will create the crd for the federated type and then fail in attempting to create the `FederatedTypeConfig` (since it already exists).   `kubefed2 enable` should check for the presence of the FTC for the type and fail immediately if it already exists, since a type can be enabled at most once for a control plane.

/kind bug

**Comment by draveness** [comment]
Is this issue still exists? I checked out the code in `enable.go` file and found out the cmd will update the `FederatedTypeConfig` if it already exists.

```go
	concreteTypeConfig := resources.TypeConfig.(*fedv1a1.FederatedTypeConfig)
	concreteTypeConfig.Namespace = namespace
	existingTypeConfig := &fedv1a1.FederatedTypeConfig{}
	err = client.Get(context.TODO(), existingTypeConfig, namespace, concreteTypeConfig.Name)
	createdOrUpdated := "created"
	if err != nil {
		if !apierrors.IsNotFound(err) {
			return errors.Wrapf(err, "Error retrieving FederatedTypeConfig %q", concreteTypeConfig.Name)
		}
		err = client.Create(context.TODO(), concreteTypeConfig)
		if err != nil {
			return errors.Wrapf(err, "Error creating FederatedTypeConfig %q", concreteTypeConfig.Name)
		}
	} else {
		existingTypeConfig.Spec = concreteTypeConfig.Spec
		err = client.Update(context.TODO(), existingTypeConfig)
		if err != nil {
			return errors.Wrapf(err, "Error updating FederatedTypeConfig %q", concreteTypeConfig.Name)
		}
		createdOrUpdated = "updated"
	}
```

**Comment by font** [comment]
This issue has been superseded by #720. I would close this but considering this issue was created after #720, I'll defer to @marun.

**Comment by marun** [comment]
@font This issue was not been superseded by #720.  If you inspect the code, you will notice that CRD creation is performed before checking if an FTC exists.  This allows creation of a duplicate CRD with non-default group/version.  A check for the FTC should instead be performed in advance of CRD creation, and enabling should fail if the group and version of the CRD that will be created don't match an existing FTC.

**Comment by font** [comment]
@marun Apologies I misread your original comment. The important keyword was `non-default` group.
